packages/zoltan2/src/algorithms/partition/Zoltan2_AlgMultiJagged.hpp

```diff
@@ -78,7013 +73,9090 @@
+    this->val = other.val;
+    this->_EPSILON = other._EPSILON;
+  }
 
-    ~uMultiSortItem(){
-        //freeArray<WT>(this->val);
-    }
+  ~uMultiSortItem() {
+    // freeArray<WT>(this->val);
+  }
 
-    void set(IT index_ ,CT count_, WT *vals_){
-        this->index = index_;
-        this->count = count_;
-        this->val = vals_;
-    }
+  void set(IT index_ ,CT count_, WT *vals_) {
+    this->index = index_;
+    this->count = count_;
+    this->val = vals_;
+  }
 
+  uMultiSortItem<IT,CT,WT> operator=(const uMultiSortItem<IT,CT,WT>& other) {
+    this->index = other.index;
+    this->count = other.count;
+    this->val = other.val;
+    return *(this);
+  }
 
-    uMultiSortItem<IT,CT,WT> operator=(const uMultiSortItem<IT,CT,WT>& other){
-        this->index = other.index;
-        this->count = other.count;
-        this->val = other.val;
-        return *(this);
+  bool operator<(const uMultiSortItem<IT,CT,WT>& other) const {
+    assert(this->count == other.count);
+    for(CT i = 0; i < this->count; ++i) {
+      // if the values are equal go to next one.
+      if(ZOLTAN2_ABS(this->val[i] - other.val[i]) < this->_EPSILON) {
+        continue;
+      }
+      // if next value is smaller return true;
+      if(this->val[i] < other.val[i]) {
+        return true;
+      }
+      // if next value is bigger return false;
+      else {
+        return false;
+      }
     }
+    // if they are totally equal.
+    return this->index < other.index;
+  }
 
-    bool operator<(const uMultiSortItem<IT,CT,WT>& other) const{
-        assert (this->count == other.count);
-        for(CT i = 0; i < this->count; ++i){
-            //if the values are equal go to next one.
-            if (ZOLTAN2_ABS(this->val[i] - other.val[i]) < this->_EPSILON){
-                continue;
-            }
-            //if next value is smaller return true;
-            if(this->val[i] < other.val[i]){
-                return true;
-            }
-            //if next value is bigger return false;
-            else {
-                return false;
-            }
-        }
-        //if they are totally equal.
-        return this->index < other.index;
-    }
-    bool operator>(const uMultiSortItem<IT,CT,WT>& other) const{
-        assert (this->count == other.count);
-        for(CT i = 0; i < this->count; ++i){
-            //if the values are equal go to next one.
-            if (ZOLTAN2_ABS(this->val[i] - other.val[i]) < this->_EPSILON){
-                continue;
-            }
-            //if next value is bigger return true;
-            if(this->val[i] > other.val[i]){
-                return true;
-            }
-            //if next value is smaller return false;
-            else //(this->val[i] > other.val[i])
-            {
-                return false;
-            }
-        }
-        //if they are totally equal.
-        return this->index > other.index;
+  bool operator>(const uMultiSortItem<IT,CT,WT>& other) const {
```

Comment:
`operator<` should suffice for sorting.

Reply:
@kddevin Recommended changes would simplify code a lot. But this code is not really changed with the PR, just cleaned up formatting, so we may want to consider as a separate PR.

Reply:
I'm not your boss -- I just think that if it doesn't happen now, it probably will never happen.

Reply:
Removed operator>. I didn't realize it was unused. Thanks.